### PR TITLE
ARROW-14616: [C++] Fix build errors on master

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

File: cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc [hunk 1]
```diff
@@ -1496,13 +1496,19 @@ TEST_F(TestUnaryArithmeticDecimal, AbsoluteValue) {
       CheckScalar(func, {ArrayFromJSON(ty, R"(["1.00", "-42.15", null])")},
```
Comment (cpcloud) [inline cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc:1496]: I think I'm just dumb, but why don't these calls also require explicit construction?

**Review by cpcloud — APPROVED** [review]
LGTM, thanks for fixing.

File: cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc (hunk 1 above)
Comment (lidavidm) [reply cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc:1496]: There's only one applicable overload (DatumVector). I'm the dumb one here, I added DatumVector below where it wasn't needed. I pushed a fix.

Comment (cpcloud) [reply cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc:1496]: Awesome.

**Review by cpcloud — APPROVED** [review]

**Pull request closed without merging**